src/mono/mono/metadata/custom-attrs.c

```diff
@@ -2056,6 +2057,57 @@ mono_custom_attrs_from_method_checked (MonoMethod *method, MonoError *error)
 	return mono_custom_attrs_from_index_checked (m_class_get_image (method->klass), idx, FALSE, error);
 }
 
+gboolean
+mono_method_get_unsafe_accessor_attr_data (MonoMethod *method, int *accessor_kind, char **member_name, MonoError *error)
+{
+	MonoCustomAttrInfo *cinfo = mono_custom_attrs_from_method_checked (method, error);
+
+	if (!cinfo)
+		return FALSE;
+
+	MonoClass *unsafeAccessor = mono_class_try_get_unsafe_accessor_attribute_class ();
+	MonoCustomAttrEntry *attr = NULL;
+
+	for (int idx = 0; idx < cinfo->num_attrs; ++idx) {
+		MonoClass *ctor_class = cinfo->attrs [idx].ctor->klass;
+		if (ctor_class == unsafeAccessor) {
+			attr = &cinfo->attrs [idx];
+			break;
+		}
+	}
+
+	if (!cinfo->cached)
+		mono_custom_attrs_free(cinfo);
+
+	if (!attr)
+		return FALSE;
+
+	MonoDecodeCustomAttr *decoded_args = mono_reflection_create_custom_attr_data_args_noalloc (m_class_get_image (attr->ctor->klass), attr->ctor, attr->data, attr->data_size, error);
+	
+	if (!is_ok (error)) {
+		mono_error_cleanup (error);
+		mono_reflection_free_custom_attr_data_args_noalloc (decoded_args);
+		return FALSE;
+	}
+
+	g_assert (decoded_args->typed_args_num == 1);
+	*accessor_kind =  *(int*)decoded_args->typed_args [0]->value.primitive;
+
+	for (int i = 0; i < decoded_args->named_args_num; ++i) {
```

Comment:
I would like to know if we can utilize the `method_has_wellknown_attribute` and `MonoHasValueCallback` logic to retrieve the value of the custom attribute.

Reply:
@kotlarmilos It's not worth while to do that here, in my opinion. The primary goal of `mono_method_metadata_foreach_custom_attr` is to traverse the custom attribute table without triggering any additional class loading - this is needed when the class loader itself needs to depend on some custom attributes (like `IsByRefLike` or `InlineArray`).  It comes with limitations - namely - we dont' do any parsing of the custom attribute value blob.  It's harder to use, so we only really use it when there's no choice.

But in this case we don't have restrictions - it's not the class loader that needs the `UnsafeAccessor` info - it's the execution engines.  So we can use the less restricted functions like `mono_reflection_create_custom_attr_data_args_noalloc`.

Reply:
Oh, I guess we can pass a callback to `mono_method_metadata_foreach_custom_attr` that parses the value blob. I forgot you added that. Sorry.  So yea I guess it's possible to use it here.  Although iterating over the `MonoCustomAttrInfo` will do the same thing

Reply:
Right. I will leave it up to @fanyang-mono to decide since that would require creating a callback.

Reply:
Which one is faster performance-wise?